modules/navier_stokes/src/userobjects/RhieChowMassFlux.C

```diff
@@ -0,0 +1,516 @@
+  params += NonADFunctorInterface::validParams();
+
+  params.addClassDescription("Computes H/A and 1/A together with face mass fluxes for segregated "
+                             "momentum-pressure equations using linear systems.");
+
+  params.addRequiredParam<VariableName>(NS::pressure, "The pressure variable.");
+  params.addRequiredParam<VariableName>("u", "The x-component of velocity");
+  params.addParam<VariableName>("v", "The y-component of velocity");
+  params.addParam<VariableName>("w", "The z-component of velocity");
+  params.addRequiredParam<std::string>("p_diffusion_kernel",
+                                       "The diffusion kernel acting on the pressure.");
+
+  params.addRequiredParam<MooseFunctorName>(NS::density, "Density functor");
+
+  // We disable the execution of this, should only provide functions
+  // for the SIMPLE executioner
+  ExecFlagEnum & exec_enum = params.set<ExecFlagEnum>("execute_on", true);
+  exec_enum.addAvailableFlags(EXEC_NONE);
+  exec_enum = {EXEC_NONE};
+  params.suppressParameter<ExecFlagEnum>("execute_on");
+
+  return params;
+}
+
+RhieChowMassFlux::RhieChowMassFlux(const InputParameters & params)
+  : GeneralUserObject(params),
+    BlockRestrictable(this),
+    NonADFunctorInterface(this),
+    _moose_mesh(UserObject::_subproblem.mesh()),
+    _mesh(_moose_mesh.getMesh()),
+    _dim(blocksMaxDimension()),
+    _p(dynamic_cast<MooseLinearVariableFVReal *>(
+        &UserObject::_subproblem.getVariable(0, getParam<VariableName>(NS::pressure)))),
+    _vel(_dim, nullptr),
+    _HbyA_flux(_moose_mesh, blockIDs(), "HbyA_flux"),
+    _Ainv(_moose_mesh, blockIDs(), "Ainv"),
+    _face_mass_flux(_moose_mesh, blockIDs(), "face_values"),
+    _rho(getFunctor<Real>(NS::density))
+{
+  if (!_p)
+    paramError(NS::pressure, "the pressure must be a MooseLinearVariableFVReal.");
+  checkBlocks(*_p);
+
+  std::vector<std::string> vel_names = {"u", "v", "w"};
+  for (const auto i : index_range(_vel))
+  {
+    _vel[i] = dynamic_cast<MooseLinearVariableFVReal *>(
+        &UserObject::_subproblem.getVariable(0, getParam<VariableName>(vel_names[i])));
+
+    if (!_vel[i])
+      paramError(vel_names[i], "the velocity must be a MOOSELinearVariableFVReal.");
+    checkBlocks(*_vel[i]);
+  }
+
+  // Register the elemental/face functors which will be queried in the pressure equation
+  for (const auto tid : make_range(libMesh::n_threads()))
+  {
+    UserObject::_subproblem.addFunctor("Ainv", _Ainv, tid);
+    UserObject::_subproblem.addFunctor("HbyA", _HbyA_flux, tid);
+  }
+
+  if (!dynamic_cast<SIMPLE *>(getMooseApp().getExecutioner()))
+    mooseError(this->name(),
+               " should only be used with a linear segregated thermal-hydraulics solver!");
+}
+
+void
+RhieChowMassFlux::linkMomentumPressureSystems(
+    std::vector<LinearSystem *> momentum_systems,
+    const LinearSystem & pressure_system,
+    const std::vector<unsigned int> & momentum_system_numbers)
```

Comment:
do you need to pass the pass the numbers? could you not get them from the vector of systesm?

Reply:
Yeah technically I could, but I already use them in the linear system so I though I could just pass them.